bot/exts/events/trivianight/_scoreboard.py

```diff
@@ -0,0 +1,186 @@
+from discord import ButtonStyle, Embed, Interaction, Member
+from discord.ui import Button, View
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+
+class ScoreboardView(View):
+    """View for the scoreboard."""
+
+    def __init__(self, bot: Bot):
+        super().__init__()
+        self.bot = bot
+
+    @staticmethod
+    def _int_to_ordinal(number: int) -> str:
+        """
+        Converts an integer into an ordinal number, i.e. 1 to 1st.
+
+        Parameters:
+            - number: an integer representing the number to convert to an ordinal number.
+        """
+        suffix = ["th", "st", "nd", "rd", "th"][min(number % 10, 4)]
+        if (number % 100) in {11, 12, 13}:
+            suffix = "th"
+
+        return str(number) + suffix
+
+    async def create_main_leaderboard(self) -> Embed:
+        """
+        Helper function that iterates through `self.points` to generate the main leaderboard embed.
+
+        The main leaderboard would be formatted like the following:
+        **1**. @mention of the user (# of points)
+        along with the 29 other users who made it onto the leaderboard.
+        """
+        formatted_string = ""
+
+        for current_placement, (user, points) in enumerate(self.points.items()):
+            if current_placement + 1 > 30:
+                break
+
+            user = await self.bot.fetch_user(int(user))
+            formatted_string += f"**{current_placement + 1}.** {user.mention} "
+            formatted_string += f"({points:.1f} pts)\n"
+            if (current_placement + 1) % 10 == 0:
+                formatted_string += "⎯⎯⎯⎯⎯⎯⎯⎯\n"
+
+        main_embed = Embed(
+            title="Winners of the Trivia Night",
+            description=formatted_string,
+            color=Colours.python_blue,
+        )
+
+        return main_embed
+
+    async def _create_speed_embed(self) -> Embed:
+        """
+        Helper function that iterates through `self.speed` to generate a leaderboard embed.
+
+        The speed leaderboard would be formatted like the following:
+        **1**. @mention of the user ([average speed as a float with the precision of one decimal point]s)
+        along with the 29 other users who made it onto the leaderboard.
+        """
+        formatted_string = ""
+
+        for current_placement, (user, time_taken) in enumerate(self.speed.items()):
+            if current_placement + 1 > 30:
+                break
+
+            user = await self.bot.fetch_user(int(user))
+            formatted_string += f"**{current_placement + 1}.** {user.mention} "
+            formatted_string += f"({(time_taken[-1] / time_taken[0]):.1f}s)\n"
+            if (current_placement + 1) % 10 == 0:
+                formatted_string += "⎯⎯⎯⎯⎯⎯⎯⎯\n"
+
+        speed_embed = Embed(
+            title="Average time taken to answer a question",
+            description=formatted_string,
+            color=Colours.python_blue
+        )
+        return speed_embed
```

Comment:
I don't remember, does this account for people not answering all questions? Also we should probably filter out incorrect answers (if possible?)